flink-kubernetes/src/main/java/org/apache/flink/kubernetes/kubeclient/resources/KubernetesSharedInformer.java

```diff
@@ -265,14 +266,32 @@ private WatchCallback(
         private void run(Consumer<WatchCallbackHandler<T>> handlerConsumer) {
             if (this.executorService == null) {
                 handlerConsumer.accept(handler);
-            } else {
-                this.executorService.submit(
+                return;
+            }
+            final int present = handling.getAndIncrement();
+            try {
+                queue.put(handlerConsumer);
+            } catch (InterruptedException e) {
+                Thread.currentThread().interrupt();
+                return;
+            }
+            if (present == 0) {
+                executorService.submit(
```

Comment:
We need to use the `execute` instead of `submit` here. Right? Because it can swallow exceptions.